pkg/events/redis.go

```diff
@@ -51,6 +52,11 @@ func NewRedisStream(sub substrate.Manager, address string, farm pkg.FarmID, node
 	}, nil
 }
 
+func (r *RedisStream) UpdateSubstrateManager(sub substrate.Manager) {
+	r.sub = sub
+	r.processor.sub = sub
```

Comment:
also here u r just changing sub and u didn't deal with the already started stream